kogito-codegen/src/main/resources/class-templates/DecisionRestResourceTemplate.java

```diff
@@ -39,11 +33,12 @@
     @io.swagger.v3.oas.annotations.responses.ApiResponse(content = @io.swagger.v3.oas.annotations.media.Content(mediaType = "application/json",schema = @io.swagger.v3.oas.annotations.media.Schema(ref = "/dmnDefinitions.json#/definitions/OutputSet1")))
     public $outputType$ dmn($inputType$ variables) {
         org.kie.kogito.decision.DecisionModel decision = application.decisionModels().getDecisionModel("$modelNamespace$", "$modelName$");
-        OutputSet outputSet = (OutputSet)StronglyTypedUtils.convertToOutputSet(variables, OutputSet.class);
+        OutputSet outputSet = (OutputSet) StronglyTypedUtils.convertToOutputSet(variables, OutputSet.class);
         org.kie.kogito.dmn.rest.DMNResult result = new org.kie.kogito.dmn.rest.DMNResult("$modelNamespace$", "$modelName$", decision.evaluateAll(DMNJSONUtils.ctx(decision, $inputData$)));
+        enrichResponseHeaders(result);
```

Comment:
I think you should swap these two lines and avoid to add `Metadata` to `rest.DMNResult`. This object is sent back to user and it should not contain other/external data
```suggestion
        enrichResponseHeaders(result);
        org.kie.kogito.dmn.rest.DMNResult result = new org.kie.kogito.dmn.rest.DMNResult("$modelNamespace$", "$modelName$", decision.evaluateAll(DMNJSONUtils.ctx(decision, $inputData$)));
```

Reply:
I like this as it would imply no modification to `DMNResult` implementations required 👍

On other hand this just go to highlight there is no clear strategy for "metadata" handling on Kogito design

Reply:
I removed that now...no need for it anymore with the latest changes :+1: